### DISCUSSION TITLE
useCookie and domain strictness

### DISCUSSION THREAD

**Discussion by rrolla** [nuxt/nuxt#19283] [Questions]
I have nuxt 3 site where I set `user` strict cookie with specific domain `.mysite.com`

And I have staging for the same site where I set cookie specific for domain `.staging.mysite.com`

Is it expected behavior for Nuxt to read my cookie with the top-level domain `.mysite.com` instead subdomain `.staging.mysite.com`?

Problem is that those cookies clashes, if I'm logged in on production, nuxt read that production cookie also on my staging ignoring my setted domain
```js
const user = useCookie('user', { domain: '.staging.mysite.com' }))
```
![Screenshot from 2023-02-25 21-26-31](https://user-images.githubusercontent.com/3932546/221376172-0f14f139-f7aa-4b1f-ac7e-5e4a74b55293.png)

**Comment by rrolla** [comment]
Does anybody else have this issue?

**Reply by drevantonder** [reply]
Yes, I run into this issue as well.

**Reply by TheAlexLichter** [reply]
Should be worth raising an issue 👍
